### ISSUE TITLE
As a 'Data Analyst', I would like to only view sections that need submission

### ISSUE THREAD

**Issue by lilybl1** [raft-tech/TANF-app#2217]
**Description:**
As a 'Data Analyst', I would like to only view sections that I need to submit data files for so that I'm not having to waste time or worry about data files that do not apply to my STT. 

We observed during our v2.0 pilot that certain STTs we held moderated sessions with did not necessarily know what to do with the Section 4 area as they did not have that data file to submit. The knowledge center does have a note about this as a stop gap. 

**Acceptance Criteria:**
- [ ] SSP/TANF radio button continues to drive updates to Reports section below search button
- [ ] Users will not longer see all Sections 1-4 but instead headers based on backend models
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary

**Tasks:**
- [x] Render the frontend reports page based on backend's model `STT.filenames.keys()` (e.g. ['Active Case Data', 'Closed Case Data', 'Aggregate Data', 'Stratum Data'])
- [x] Ensure frontend unit tests are updated to account for this more dynamic shift for TANF, SSP, and Tribal
- [x] If necessary, ensure the backend filenames.keys are passed to frontend

11/1: Discussed that we would continue to keep the radio button for TANF and SSP to help focus the user on each type

**Notes:**
Scenario: TANF Data Files
Given the STT Data Analyst has been **approved** to submit certain TANF Data Files (Section 1, 2, 3 and/or 4)
When logged into TDP
Then display all applicable sections (Active, Closed, Aggregate and/or Stratum) that the STT Data Analyst has been approved to submit

![image](https://user-images.githubusercontent.com/28515874/202292998-6e752965-aeda-4ed4-9515-5cdce26fb30e.png)
https://www.figma.com/file/irgQPLTrajxCXNiYBTEnMV/TDP-Mockups-For-Feedback?node-id=7624%3A50249&t=XjNrtEH5J4odxe7L-1

Scenario: SSP Data Files
Given the STT Data Analyst has been **approved** to submit certain STT Data Files (Section 1, 2, 3 and/or 4)
When logged into TDP
Then display all applicable sections (Active, Closed, Aggregate and/or Stratum) that the STT Data Analyst has been approved to submit

**Comment by lilybl1** [comment]
Discussed in Backlog Refinement that this ticket should be generic not specific to Section 4.

**Comment by lilybl1** [comment]
@andrew-jameson Updated per our discussion - can we sync on the triggers for the other sections?

**Comment by stevenino** [comment]
To be re-scoped to include all data types.

**Comment by andrew-jameson** [comment]
Self note: Write dev-ready requirements for frontend to display "keys" from data/states.csv for example to no longer show "section 1", "section 2" but instead selectively show "active", "closed", "aggregate" based on what they are supposed to submit.

**Comment by andrew-jameson** [comment]
@reitermb If you can find any figma mockups for this, please do add them under notes otherwise we will proceed as-is.

**Comment by reitermb** [comment]
> @reitermb If you can find any figma mockups for this, please do add them under notes otherwise we will proceed as-is.

We didn't have any that were up to date, but I spun one up for the sake of illustration. cc @stevenino

**Comment by stevenino** [comment]
@andrew-jameson Are figma mockups required? I know we wanted them if they already existed but I didn't think we needed to make new ones if there were none.

**Comment by andrew-jameson** [comment]
> @andrew-jameson Are figma mockups required? I know we wanted them if they already existed but I didn't think we needed to make new ones if there were none.

No, not required. Good to have, thanks for adding to description, @reitermb

**Comment by reitermb** [comment]
> @andrew-jameson Are figma mockups required? I know we wanted them if they already existed but I didn't think we needed to make new ones if there were none.

Just wanted to note that this was ~a 30 second modification on existing figma assets so I figured I'd just go for it

**Comment by stevenino** [comment]
Perfect. Thank you!

**Comment by lhuxraft** [comment]
11/22: Almost ready for Raft review. Failing code coverage, adding more tests

**Comment by lhuxraft** [comment]
11/25: Demoed on office hours on Friday, made suggested changes.

**Comment by lhuxraft** [comment]
12/4: Adjusting based on changes, will request re-review once complete
